# HG changeset patch
# User Rik <rik@octave.org>
# Date 1540937902 25200
#      Tue Oct 30 15:18:22 2018 -0700
# Node ID 07d14d092d773b7b5c6ccba8faeb55b2823c8c57
# Parent  49ffd0e0242de71a8b61c7ab7b91b6f8b3c322c2
hgsave.m: Validate there is only a single graphics handle input.

* hgsave.m: Verify that graphics handle input H is a scalar.  Add BIST test for
same.

diff --git a/scripts/plot/util/hgsave.m b/scripts/plot/util/hgsave.m
--- a/scripts/plot/util/hgsave.m
+++ b/scripts/plot/util/hgsave.m
@@ -58,22 +58,26 @@ function hgsave (h, filename, fmt = "-bi
     print_usage ();
   endif
 
   ## Check input arguments
   if (nargin == 1 && ischar (h))
     filename = h;
     h = get (0, "currentfigure");
     if (isempty (h))
-      error ("hgsave: No current figure to save");
+      error ("hgsave: no current figure to save");
     endif
   elseif (! (ishghandle (h) && ischar (filename)))
     print_usage ();
   endif
 
+  if (! isscalar (h))
+    error ("hgsave: H must be a single graphics handle");
+  endif
+
   ## Check file extension
   [~, ~, ext] = fileparts (filename);
   if (isempty (ext))
     filename = [filename ".ofig"];
   endif
 
   s_oct40 = hdl2struct (h);
   save (fmt, filename, "s_oct40");
@@ -122,8 +126,16 @@ endfunction
 %!   close (h2);
 %!   graphics_toolkit (toolkit);
 %! end_unwind_protect
 
 ## Test input validation
 %!error hgsave ()
 %!error hgsave (1, 2, 3, 4)
 %!error hgsave ("abc", "def")
+%!error <H must be a single graphics handle>
+%! unwind_protect
+%!   hf = figure ("visible", "off");
+%!   hax = axes ();
+%!   hgsave ([hf, hax], "foobar");
+%! unwind_protect_cleanup
+%!   close (hf);
+%! end_unwind_protect
